src/sentry/dynamic_sampling/rules_generator.py

```diff
@@ -54,6 +59,26 @@ def generate_environment_rule() -> BaseRule:
     }
 
 
+def generate_healthcheck_rule(sample_rate: float) -> BaseRule:
+    return {
+        "sampleRate": sample_rate / HEALTH_CHECK_DROPPING_FACTOR,
+        "type": "trace",
+        "condition": {
+            "op": "or",
+            "inner": [
+                {
+                    "op": "glob",
+                    "name": "transaction.transaction",
```

Comment:
`transaction.transaction` doesn't match, the names are either `trace.transaction` or `event.transaction`. Since the type of the rule is `trace`, I assume you're interested in `trace.transaction`, see [in relay](https://github.com/getsentry/relay/blob/80a6dd9bce1fea7d1df1453332cbae6f382b3c15/relay-sampling/src/lib.rs#L834-L839).

Same for the tests.

Reply:
cc @jan-auer  suggested `transaction.transaction` from https://github.com/getsentry/relay/blob/a9355f574060b0bcaad369daa2ed0bd9218709ad/relay-sampling/src/lib.rs#L372-L375